libs/defi/oeth/src/redeem/components/Swapper.tsx

```diff
@@ -118,91 +119,76 @@ export const Swapper = ({
       }}
       onApproveFailure={(state) => {
         const { error, trackId } = state;
-        updateActivity({
+        const updated = updateActivity({
           ...state,
```

Comment:
This may throw in some unexpected content in there by passing everything from state inside? This comment applies to anywhere this is done.